### DISCUSSION TITLE
Status 404 after update

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2428] [Help] [answered]
After updating the budibase setup from 0.9.107 to 0.9.111 i cannot reach the database anymore via 
mydomain.com/app/<app-name>/

budibase: 0.9.111
docker-compose on ubuntu
served with caddy 
app based on budibase csv

the app can be called from the portal, however i do then have the long numberstring as URL instead of the short Domainname.
When i try to open it via `mydomain.com/app/<app-name>/ ` i get 

`{"message":"Database does not exist.","status":404}`

is there anything i missed while updating?

**Comment by sergey-dme** [comment]
Hey! Did you try to clear cookies?

**Reply by tobehn** [reply]
I just did without success..

**Comment by shogunpurple** [comment]
@tobehn if you have been using budibase for a while this sounds like you may need to update your `envoy.yaml` file from our repository. After updating this, you will need to restart your `envoy-proxy` in your environment.

```
# go to wherever budibase is installed
rm -rf envoy.yaml
wget https://raw.githubusercontent.com/Budibase/budibase/master/hosting/envoy.yaml
docker-compose restart proxy-service
```

Then try again. I think that should resolve it!

**Reply by tobehn** [reply]
@shogunpurple thanks for your advice.
Sadly i could not solve it by this. Budibase was setup with version 0.9.97, updated to 0.9.107 once and then to 0.9.111 again. I always also updated the app itself to the used versions of budibase.

I use Caddy to serve the app via reverse proxy and my own domain (like written in your docs). 
I can reach the app via its "cryptic" (.../app_dev_) domain.

**Comment by tobehn** [comment]
I could not really solve this issue by the suggested procedures.

My work arround:

- exported the app
- deleted it
- and imported it with the same name

**Comment by tobehn** [comment]
Sadly i get the same error again.
It was running served by caddy. I the changed a single thing. i unchecked the line graph to not animate. Published and then the app was not accessible anymore.

I think beforehand i updated from 0.9.1.111 to 112 and converted the app. Does the update process somehow make the app inaccessible via the short link with the apps name?



`


Accept | text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
-- | --
Accept-Encoding | gzip, deflate, br
Accept-Language | de,en-US;q=0.7,en;q=0.3
Cache-Control | max-age=0
Connection | keep-alive
DNT | 1
Host | bb.exampledomain.io
Sec-Fetch-Dest | document
Sec-Fetch-Mode | navigate
Sec-Fetch-Site | none
Sec-Fetch-User | ?1
Sec-GPC | 1
TE | trailers
Upgrade-Insecure-Requests | 1
User-Agent | Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0

`

**Reply by shogunpurple** [reply]
This seems like a problem with the reverse proxy given that it works with the normal domain. Would it be possible to look at your caddy configuration?

**Reply by tobehn** [reply]
well, its just a `Caddyfile` with the domain (Host) in it. I did not configure anything else. 
I basically did what i could read in the docs (https://docs.budibase.com/self-hosting/advanced-options/reverse-proxy)

**Reply by shogunpurple** [reply]
@tobehn is this still a problem on the latest release?

**Reply by tobehn** [reply]
@shogunpurple i just updated from 0.9.114 to 116 by hitting update button in the web app in order to check
However this led to total loss - i am now redirected to /builder/admin and asked to create a new admin account.
Is there a way of getting the data back? in the docker-compose are there defined any persitent volumes?

BB is installed on an Ubuntu vps machine served with Caddy.

basically with 

- hosting.properties
- envoy.yaml
- docker-compose.yaml 

in /opt/ and started from there
this might be another issue though

**Reply by shogunpurple** [reply]
Hey @tobehn - don't worry, you won't have lost your data.

I think this is the `envoy.yaml` issue again above https://github.com/Budibase/budibase/discussions/2428#discussioncomment-1219485

Run those commands and see if it gets you back to the portal.

**Reply by tobehn** [reply]
@shogunpurple aww got you. I checked the master repo for changes in envoy.yaml and couldn't see any since 26days. But i guess i have the latest now. So all User data is back.

But the 404 is still the same.
After this update i cannot access the web via its name link app/appname anymore. Its "missing" 404.

Is there any other information i can provide? Let me know.

**Reply by shogunpurple** [reply]
Hey @tobehn - we've identified the issue and it will be in the next release this afternoon. Thanks a lot for your patience!

**Reply by tobehn** [reply]
@shogunpurple great. I can definitely perform an update later this day and check it again.
Do I need to recreate my app like descibed [above](https://github.com/Budibase/budibase/discussions/2428#discussioncomment-1221716). Or do you think it can be left like that and the update will solve the issue?

**Reply by shogunpurple** [reply]
Nope, your app should be fine and the update should solve the issue 👍

**Reply by tobehn** [reply]
I can absolutely confirm that you fixed it, just hit the update and smoothly made it to 117. Apps are accessible.


@shogunpurple, @mjashanks , @joebudi, @mike12345567, @aptkingston and @kevmodrome as well as all the contibutors:

Thank you for this **beautiful** piece of software.

**Reply by shogunpurple** [reply]
Glad to hear that! Thank you so much for your kind words @tobehn 🙏

**Comment by tobehn** [comment]
@shogunpurple sadly i just ran into the same issue. Whereas i do not really know which update led to it. 
I updated to 0.9.122 yesterday, converted a few apps and again i cannot reach them via their name.

Did you somehow revert the changes that made the update process working?

**Comment by tobehn** [comment, marked as answer]
@shogunpurple - gosh i finally found out what caused the 404 missing database. 

It is authentication is not given. When i am logged in i of course can access the short link of an app via the address bar. 
In case i am not logged in and directly want to access via URL i get the 404.

This should then redirect to the login in my opinion. Do you think that is possible? See #2835

**Comment by jhina9238-ship-it** [comment]
United ff mod
